### PR TITLE
Finalizing dex.trades_beta

### PR DESCRIPTION
**Pull request by Hosuke**
In this PR:
- add block_number in [oneinch_lop_own_trades](https://github.com/duneanalytics/spellbook/pull/5303/commits/41382b16f1aa6c447626803fae5d1c954bac9cff)
- add block_number in [zeroex_native_trades](https://github.com/duneanalytics/spellbook/pull/5303/commits/200a6284d36368d42954d2d33e080495ffe604be)
- union oneinch_lop_own_trades and zeroex_native_trades directly into dex.trades_beta
- rewrite union columns in full list

[dex.trades_beta coverage checking](https://dune.com/queries/3424832)

### CODE REVIEW

File: models/zeroex/zeroex_native_trades.sql
Comment (Hosuke) [inline models/zeroex/zeroex_native_trades.sql:1]: cc. @RantumBits for review.

File: models/oneinch/oneinch_lop_own_trades.sql
Comment (Hosuke) [inline models/oneinch/oneinch_lop_own_trades.sql:1]: cc. @max-morrow for review.

Comment (jeff-dude) [reply models/oneinch/oneinch_lop_own_trades.sql:1]: @grkhr may want to have a look as well

File: models/_sector/dex/trades/dex_trades_beta.sql [hunk 1]
```diff
@@ -32,12 +38,76 @@ WITH curve AS (
...
+    WHERE
+        {{ incremental_predicate('block_time') }}
+    {% endif %}
+    {% if not loop.last %}
+    UNION ALL
+    {% endif %}
+    {% endfor %}
+)
+
+{% set cte_to_union = ['curve', 'as_is_dexs', 'dexs'] %}
```
Comment (jeff-dude) [inline models/_sector/dex/trades/dex_trades_beta.sql:80]: nice usage on the union of CTEs here

minor, but can you make the list one value per line?

```suggestion
{% set cte_to_union = [
    'curve'
    , 'as_is_dexs'
    , 'dexs'
    ] 
%}
```

**Review by jeff-dude — CHANGES_REQUESTED** [review]
one minor comment below. i like keeping this pretty simple for 1inch / 0x lineage. i think there is room to improve 0x design, but that's for another day

File: models/oneinch/oneinch_lop_own_trades.sql
Comment (grkhr) [reply models/oneinch/oneinch_lop_own_trades.sql:1]: approve from us!

**Review by jeff-dude — APPROVED** [review]
updated the curve macro to use new `amount_usd` logic

this lgtm 🙏

**Pull request merged**